pushnotifications/src/main/java/com/pusher/pushnotifications/internal/ServerSyncJob.kt

```diff
@@ -1,35 +1,44 @@
 package com.pusher.pushnotifications.internal
 
 import com.pusher.pushnotifications.api.DeviceMetadata
-import com.squareup.moshi.adapters.PolymorphicJsonAdapterFactory
+import com.squareup.moshi.JsonClass
+import dev.zacsweers.moshisealed.annotations.TypeLabel
 import java.io.Serializable
 
-sealed class ServerSyncJob: Serializable {
-
-    companion object {
-        // the following is required for the Persistent Job Queue to read and write data correctly.
-        // if you add another ServerSyncJob, you must also add it below here too!
-        val jsonAdapterFactory: PolymorphicJsonAdapterFactory<ServerSyncJob>
-                = PolymorphicJsonAdapterFactory.of(ServerSyncJob::class.java, "ServerSyncJob")
-                .withSubtype(StartJob::class.java, "StartJob")
-                .withSubtype(RefreshTokenJob::class.java, "RefreshTokenJob")
-                .withSubtype(SubscribeJob::class.java, "SubscribeJob")
-                .withSubtype(UnsubscribeJob::class.java, "UnsubscribeJob")
-                .withSubtype(SetSubscriptionsJob::class.java, "SetSubscriptionsJob")
-                .withSubtype(ApplicationStartJob::class.java, "ApplicationStartJob")
-                .withSubtype(SetUserIdJob::class.java, "SetUserIdJob")
-                .withSubtype(StopJob::class.java, "StopJob")
-    }
-}
+@JsonClass(generateAdapter = true, generator = "sealed:ServerSyncJob")
+sealed class ServerSyncJob: Serializable { }
```

Comment:
`Serializable` marker looks redundant and if it is indeed redundant it is also confusing and error-prone.

Reply:
hmm. the code doesn't compile if I remove this. The ServerSyncHandler has a JobQueue ( which expects the types to be Serializable (https://github.com/pusher/push-notifications-android/blob/15f253d5ade2d783cd8aae52763de3c0bbcfcd11/pushnotifications/src/main/java/com/pusher/pushnotifications/internal/PersistentJobQueue.kt). I think this would require more refactoring to remove.

Reply:
Oh I didn't expect serialisation is being done using two mechanisms. Thanks taking the feedback onboard and for checking! Yeah refactoring of that is not valuable enough to give it a shot.